src/LuaTransformer.ts

```diff
@@ -376,8 +376,26 @@ export class LuaTransformer {
             throw TSTLErrors.UnsupportedImportType(statement.importClause);
         }
 
-        const type = this.checker.getTypeAtLocation(imports);
-        const shouldResolve = !tsHelper.getCustomDecorators(type, this.checker).has(DecoratorKind.NoResolution);
+        let shouldResolve = true;
+        if (ts.isNamedImports(imports)) {
+            for (const importSpecifier of imports.elements) {
```

Comment:
I don't think we can rely on elements there. First, what about `import {} from "fake"`? Second, I'm not sure if that would work with re-exporting.
We also have to consider a case when we have no elements to check: `import "fake"`.
I think we should check out the actual type of module, not types of imported elements. I'm not really sure how to do it, but editors that use TSLS seem to have resolved module path on hover, so type checker should have it

Reply:
Thanks for the hand! I can't think of a way to test `import {} from "fake"` but I've got `import "fake"` validated by two new tests.

Also using `this.checker.getSymbolAtLocation(statement.moduleSpecifier)` looks much more reliable